jupyter_core/command.py

```diff
@@ -113,9 +115,24 @@ def _execvp(cmd, argv):
         os.execvp(cmd, argv)
 
 
+def _jupyter_realpath(path, subcommand):
+    # get currrent PATH
+    _path_env = os.environ['PATH']
+    # set new PATH with self
+    os.environ['PATH'] = os.pathsep.join(_path_with_self())
+    # get the real path for the jupyter-<subcommand>
+    real_path = find_executable('{}-{}'.format(path, subcommand))
```

Comment:
shutil.which is not available for py2

Reply:
But good news, this repo already has a backport - `from .utils.shutil_which import which`. This is a bit more complex than `find_executable`, e.g. it handles PATHEXT on Windows, so it would be good to use that (and the real `shutil.which` on Python 3, of course).